pkg/resource-handler/controller/shard/pool_service.go

```diff
@@ -16,16 +16,17 @@ const (
 	PoolComponentName = "shard-pool"
 )
 
-// BuildPoolHeadlessService creates a headless Service for a pool's StatefulSet.
+// BuildPoolHeadlessService creates a headless Service for a pool's StatefulSet in a specific cell.
 // Headless services are required for StatefulSet pod DNS records.
 func BuildPoolHeadlessService(
 	shard *multigresv1alpha1.Shard,
 	poolName string,
-	poolSpec multigresv1alpha1.ShardPoolSpec,
+	cellName string,
+	poolSpec multigresv1alpha1.PoolSpec,
 	scheme *runtime.Scheme,
 ) (*corev1.Service, error) {
-	name := buildPoolName(shard.Name, poolName)
-	labels := buildPoolLabels(shard, poolName, poolSpec)
+	name := buildPoolNameWithCell(shard.Name, poolName, cellName)
+	labels := buildPoolLabelsWithCell(shard, poolName, cellName, poolSpec)
```

Comment:
You changed this but the declaration for buildPoolLabels remains and it is unused.